openmc/mgxs_library.py

```diff
@@ -0,0 +1,705 @@
+        number of points to use, if ``enable`` is True.
+
+    """
+    def __init__(self, name, energy_groups, representation="isotropic"):
+        # Initialize class attributes
+        self._name = name
+        self._energy_groups = energy_groups
+        self._representation = representation
+        self._alias = None
+        self._kT = None
+        self._fissionable = False
+        self._scatt_type = 'legendre'
+        self._order = None
+        self._tabular_legendre = None
+        self._num_polar = None
+        self._num_azimuthal = None
+        self._total = None
+        self._absorption = None
+        self._scatter = None
+        self._multiplicity = None
+        self._fission = None
+        self._nu_fission = None
+        self._k_fission = None
+        self._chi = None
+        self._use_chi = None
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def energy_groups(self):
+        return self._energy_groups
+
+    @property
+    def representation(self):
+        return self._representation
+
+
+    @property
+    def alias(self):
+        return self._alias
+
+    @property
+    def kT(self):
+        return self._kT
+
+    @property
+    def scatt_type(self):
+        return self._scatt_type
+
+    @property
+    def order(self):
+        return self._order
+
+    @property
+    def tabular_legendre(self):
+        return self._tabular_legendre
+
+    @property
+    def num_polar(self):
+        return self._num_polar
+
+    @property
+    def num_azimuthal(self):
+        return self._num_azimuthal
+
+    @property
+    def total(self):
+        return self._total
+
+    @property
+    def absorption(self):
+        return self._absorption
+
+    @property
+    def scatter(self):
+        return self._scatter
+
+    @property
+    def multiplicity(self):
+        return self._multiplicity
+
+    @property
+    def fission(self):
+        return self._fission
+
+    @property
+    def nu_fission(self):
+        return self._nu_fission
+
+    @property
+    def k_fission(self):
+        return self._k_fission
+
+    @property
+    def chi(self):
+        return self._chi
+
+    @property
+    def num_orders(self):
+        if (self._order is not None) and (self._scatt_type is not None):
+            if self._scatt_type is 'legendre':
+                return self._order + 1
+            else:
+                return self._order
+
+    @name.setter
+    def name(self, name):
+        check_type('name for Xsdata', name, basestring)
+        self._name = name
+
+    @energy_groups.setter
+    def energy_groups(self, energy_groups):
+        # Check validity of energy_groups
+        check_type("energy_groups", energy_groups, EnergyGroups)
+
+        # Check that there is one or more groups
+        if ((energy_groups.num_energy_groups.num_group is None) or
+            (energy_groups.num_energy_groups.num_group < 1)):
```

Comment:
Absolutely correct.  Not sure where the above came from, must have been in an older rev.